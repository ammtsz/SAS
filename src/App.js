import React, { useEffect } from "react";
import AppRoutes from "./App.routes";

import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCredentials } from "./redux/user/user.selectors";
import { actionSetUserAuth } from "./redux/user/user.actions";
import { selectQuizActive } from "./redux/quiz/quiz.selectors";

const mapStateToProps = createStructuredSelector({
  user: selectCredentials,
  quizActive: selectQuizActive,
});

const mapDispatchToProps = (dispatch) => ({
  setUserAuth: () => dispatch(actionSetUserAuth()),
});

const App = (props) => {
  let { setUserAuth } = props;
  useEffect(() => {
    setUserAuth();
    return () => setUserAuth();
  }, [setUserAuth]);

  return <AppRoutes {...props} />;
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
