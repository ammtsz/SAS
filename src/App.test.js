import React from "react";
import App from "./App";

import { render, cleanup } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

import { MemoryRouter } from "react-router-dom";

import {
  actionSetUserAuth,
  actionGetUserTheme,
} from "./redux/user/user.actions";

describe("Component <App />", () => {
  const mockStore = configureMockStore();

  const store = mockStore({
    user: {
      datas: null,
      theme: "light",
    },
    quiz: {
      active: false,
    },
    report: {
      category: null,
    },
  });
  store.dispatch = jest.fn();

  let component;

  beforeEach(() => {
    component = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );
  });

  afterEach(() => {
    cleanup();
  });

  it("should render App", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it("shoud fire methods called on useEffect()", () => {
    expect(store.dispatch).toHaveBeenCalledTimes(2);
    expect(store.dispatch).toHaveBeenCalledWith(actionSetUserAuth());
    expect(store.dispatch).toHaveBeenCalledWith(actionGetUserTheme());
  });
});
