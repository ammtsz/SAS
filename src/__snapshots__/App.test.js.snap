// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <App /> should render App 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <nav
        class="sc-jSgupP dglkrd"
      >
        <h3
          class="sc-gKsewC kqPbG"
        >
          Teste Dev Frontend
        </h3>
        <div
          class="sc-iBPRYJ knSzVm"
        >
          <p
            class="sc-fubCfw ehTKJF"
          />
          <button
            class="sc-pFZIQ fVoRDH"
          >
            Log in
          </button>
          <span
            class="sc-jrAGrp cnBeoq"
          >
            <div
              class="sc-kEjbxe jyXxIT"
            >
              <label
                class="sc-iqHYGH eXtfx"
              >
                <input
                  class="sc-dQppl gXvpZz"
                  id="button-switch--navbar-toggle-theme"
                  type="checkbox"
                />
                <span
                  class="sc-crrsfI ejgHsP slider round"
                />
              </label>
              <label
                class="button-switch--label"
                for="button-switch--navbar-toggle-theme"
              >
                light
              </label>
            </div>
          </span>
        </div>
      </nav>
    </header>
    <main
      class="sc-bqyKva elOEp"
    >
      <div
        class="sc-hKgILt kCQzmP"
      >
        <div
          class="sc-eCssSg elQFSQ"
        />
      </div>
    </main>
  </div>
</DocumentFragment>
`;
